core/foundation/inc/ROOT/RError.hxx

```diff
@@ -189,15 +189,16 @@ RResult<T>::Unwrap() can be used as a short hand for
 int value = FuncThatReturnsRResultOfInt().Unwrap();  // may throw
 ~~~
 
-There is no implict operator that converts RResult<T> to T. This is intentional to make it clear in the calling code
+There is no implicit operator that converts RResult<T> to T. This is intentional to make it clear in the calling code
 where an exception may be thrown.
 */
 // clang-format on
 template <typename T>
 class RResult : public RResultBase {
 private:
-   /// The result value in case of successful execution
-   T fValue;
+   /// The result value in case of successful execution.
+   // Note that if T is not default constructible we wrap it in an optional to allow for an "empty" value.
+   std::conditional_t<std::is_default_constructible_v<T>, T, std::optional<T>> fValue;
```

Comment:
Do we maybe want to just always wrap it in a `std::optional`? That would simplify the code and convey the right semantics...

Reply:
That would be suboptimal for the majority of cases (default constructible `T`) and imho not really justified given how little code is necessary to support the conditional `optional`.

Reply:
Indeed, even a default constructor can be expensive. +1 for just always use `std::optional`.